lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -270,6 +270,30 @@ static void LoadSwiftFormatters(lldb::TypeCategoryImplSP swift_category_sp) {
       "Swift.UnsafeMutablePointer",
       ConstString("^Swift.UnsafeMutablePointer<.+>$"), basic_synth_flags, true);
 
+  AddCXXSummary(
+      swift_category_sp,
+      lldb_private::formatters::swift::UnsafeBufferPointerSummaryProvider,
+      "Swift.UnsafeBufferPointer",
+      ConstString("^Swift.UnsafeBufferPointer<.+>$"), summary_flags, true);
+  AddCXXSynthetic(swift_category_sp,
+                  lldb_private::formatters::swift::
+                      UnsafeBufferPointerSyntheticFrontEndCreator,
+                  "Swift.UnsafeBufferPointer",
+                  ConstString("^Swift.UnsafeBufferPointer<.+>$"),
+                  basic_synth_flags, true);
+  AddCXXSummary(
+      swift_category_sp,
+      lldb_private::formatters::swift::UnsafeBufferPointerSummaryProvider,
+      "Swift.UnsafeMutableBufferPointer",
+      ConstString("^Swift.UnsafeMutableBufferPointer<.+>$"), summary_flags,
+      true);
+  AddCXXSynthetic(swift_category_sp,
+                  lldb_private::formatters::swift::
+                      UnsafeBufferPointerSyntheticFrontEndCreator,
+                  "Swift.UnsafeMutableBufferPointer",
+                  ConstString("^Swift.UnsafeMutableBufferPointer<.+>$"),
+                  basic_synth_flags, true);
+
```

Comment:
I wonder if you can use some regex magic and reduce the duplication here given they both dispatch the same version.

Reply:
We could do that but that would make the description less accurate, for instance, If I registered a single formatter for both the Immutable and Mutable using regexes, they would both show up as "Swift.UnsafeBufferPointer". I'm not sure how important is that description.

Reply:
does the description matter though? We could change the description string to be `Swift.Unsafe(|Mutable)BufferPointer` if it's only used for displaying in `type summary`

Reply:
The description doesn't quite matter because, as pointed out by Fred, it's used only on `type summary`. The latter is only used to understand which formatter actually triggered, so, in this case, given the formatter triggering is exactly the same, it won't matter. My $0.02.